src/ducks/components/App.jsx

```diff
@@ -37,34 +38,36 @@ export class App extends Component {
     const defaultPart = enabledPages ? enabledPages[0] : 'discover'
     return (
       <BreakpointsProvider>
-        <Layout>
-          {flag('switcher') && <FlagSwitcher />}
-          <Alerter />
-          <Sidebar />
-          <Main>
-            <Switch>
-              <Route path="/redirect" component={IntentRedirect} />
-              {enabledPages.map(name => {
-                if (componentsMap[name]) {
-                  return (
-                    <Route
-                      path={`/${name}`}
-                      component={componentsMap[name]}
-                      key={name}
-                    />
-                  )
-                }
-              })}
-              {defaultPart && (
-                <Fragment>
-                  <Redirect exact from="/" to={`/${defaultPart}`} />
-                  <Redirect from="*" to={`/${defaultPart}`} />
-                </Fragment>
-              )}
-            </Switch>
-          </Main>
-          <IconSprite />
-        </Layout>
+        <WebviewIntentProvider>
```

Comment:
i don't think the PR is ready for review. Tests are failing :(

Reply:
Yes, I did not edit tests yet. Those are failing because one test is still checking for old mechanism.

The new implementation can still be reviewed for feedback, it should not change a lot.

Reply:
Tests are implemented and CI is green 👍